lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,457 @@
+    relative_name=dict(type='str'),
+    ttl=dict(type='int')
+)
+
+monitor_config_spec = dict(
+    profile_monitor_status=dict(type='str'),
+    protocol=dict(type='str'),
+    port=dict(type='int'),
+    path=dict(type='str'),
+    interval=dict(type='int'),
+    timeout=dict(type='int'),
+    tolerated_failures=dict(type='int')
+)
+
+
+class AzureRMTrafficManagerProfile(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                default='global'
+            ),
+            profile_status=dict(
+                type='str',
+                default='enabled',
+                choices=['enabled', 'disabled']
+            ),
+            routing_method=dict(
+                type='str',
+                default='performance',
+                choices=['performance', 'priority', 'weighted', 'geographic']
+            ),
+            dns_config=dict(
+                type='dict',
+                options=dns_config_spec
+            ),
+            monitor_config=dict(
+                type='dict',
+                default=dict(
+                    protocol='HTTP',
+                    port=80,
+                    path='/'
+                ),
+                options=monitor_config_spec
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.tags = None
+        self.location = None
+        self.profile_status = None
+        self.routing_method = None
+        self.dns_config = None
+        self.monitor_config = None
+        self.endpoints_copy = None
+
+        self.results = dict(
+            changed=False
+        )
+
+        super(AzureRMTrafficManagerProfile, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                           supports_check_mode=True,
+                                                           supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        to_be_updated = False
+
+        if not self.dns_config:
+            self.dns_config = dict(
+                relative_name=self.name,
+                ttl=60
+            )
+
+        if not self.location:
+            self.location = 'global'
+
+        response = self.get_traffic_manager_profile()
+
+        if self.state == 'present':
+            if not response:
+                to_be_updated = True
+            else:
+                self.results = response
```

Comment:
This would return all the fields and not our curated ones, this needs to be shortened.

Reply:
fixed